erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -39,9 +39,8 @@ def validate_for_items(self):
 		for d in self.get('items'):
 			check_list.append(cstr(d.item_code))
 
-			if (frappe.db.get_value("Item", d.item_code, "is_stock_item")==1 or
-				self.has_product_bundle(d.item_code)) and not d.warehouse:
-					frappe.throw(_("Reserved warehouse required for stock item {0}").format(d.item_code))
+			if frappe.db.get_value("Item", d.item_code, "is_stock_item") and not d.warehouse:
```

Comment:
If child item of a Product Bundle is stock-item, then make reserve warehouse mandatory.